### DISCUSSION TITLE
UI for NextJs

### DISCUSSION THREAD

**Discussion by dhawal-793** [JasonDsouza212/free-hit#241] [Ideas]
# Suggestions for Nextjs Directory Structure and UI
 As I am working to migrate the app to nextjs, There are some changes which I suggest.
If anyone have better suggestion, please share your thoughts.

### The directory structure will look like this:
- home page  : /
- category page : /[dynamic value]  ex.(ai,tools, etc)
- search page : /search?term=[searchvalue]
- about page : /about

### There are some more features: 
- Not found page for any invalid url.
- A loading UI while data is loading.

What's your thoughts on this @JasonDsouza212

**Comment by JasonDsouza212** [comment]
@dhawal-793 maintain the same design while migrating to next.js

**Reply by dhawal-793** [reply]
> @dhawal-793 maintain the same design while migrating to next.js

The designs will be same but the routes will be different, in the current version the app only have home page and the content changes accordingly
But in nextjs we can achieve this with different routes and the layout on all the pages will be same and we can pass the appropriate data so it will look like it is a different page but actually it is not.

Also it will be good for SEO.

Once I finish it I will attach the screenshots, then it will be clear.

**Reply by dhawal-793** [reply]
@JasonDsouza212 
 This is how the website will look in nextjs https://next-hit.vercel.app/

I have only implemented the home page but all the routes are working.
You can try it and suggest any updates if you want

**Reply by dhawal-793** [reply]
I will try to complete it by this weekend. and after that if we have any more feature then we can add

**Reply by dhawal-793** [reply]
Update: 
- Added Category Pages
- Search Segustions
- Search Page
- Fixed some styling 
- Fixed Links

**Reply by JasonDsouza212** [reply]
> @JasonDsouza212 This is how the website will look in nextjs https://next-hit.vercel.app/
> 
> I have only implemented the home page but all the routes are working. You can try it and suggest any updates if you want

The suggestion box is not working, and the tools are not being filtered when the user searches for them.

**Reply by dhawal-793** [reply]
> > @JasonDsouza212 This is how the website will look in nextjs https://next-hit.vercel.app/
> > I have only implemented the home page but all the routes are working. You can try it and suggest any updates if you want
> 
> The suggestion box is not working, and the tools are not being filtered when the user searches for them.

Try it now.

The way I implemented it, tools will not filter while typing, instead when someone searches for any letter or keyword, only then the related tools will be shown.


If you want I will remove this and implement the previous functionality.

**Comment by dhawal-793** [comment]
## Hey @JasonDsouza212  
#### Updates regarding the NextJS version:
#### Veiw the latest version here: https://next-hit.vercel.app

### Updates:
* Updated Search functionality as the current version.
* Added Community page.
* Updated Bookmark Page.
* Added different views feature.
* Added sort functionality.
* Updated Navbar look.
* Updated Sidebar look.
* Added max width of 2220px to the body.


### Pending:
* About Page
* Footer.
* List view.
* Card view style.

**Comment by dhawal-793** [comment]
## LATEST UPDATES

* Added Footer.
* Added List View.
* Updated Card View.
* Added SEO Support.
* Added Infinite Scroll.
* Added and Updated Viewports to make the website more responsive.
* Updated Search Functionality.
* Used Context API and UseReducer hooks to provide global states to the App.

View [here](https://next-hit.vercel.app).



## PENDING

* Abou Page.
* Layout for Not Found Page.
* MetaData and SEO for Dynamic Pages

**Reply by dhawal-793** [reply]
@JasonDsouza212  Please give feedback, is it fine or not, or do we even want to work on the next js version or not.

**Comment by akanksha984** [comment]
Hey @JasonDsouza212 https://github.com/JasonDsouza212/free-hit/pull/1412 Can you please check this, if this goes well ?
